x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -678,6 +684,45 @@ private List<Attribute> resolveUsingColumns(List<Attribute> cols, List<Attribute
             return resolved;
         }
 
+        private LogicalPlan resolveInsist(Insist insist, List<Attribute> childrenOutput, IndexResolution indexResolution) {
+            return insist.withAttributes(
+                insist.insistedAttributes().stream().map(a -> resolveInsistAttribute(a, childrenOutput, indexResolution)).toList()
+            );
```

Comment:
Nit: the stream styles makes it difficult to debug field resolution vs the traditional loop style used in the other methods:
```
List<Attribute> resolved = new ArrayList<>(insist.attributes().size());
for (Attribute r : insist.attributes()) {
     resolved.add(maybeResolve()...);
}
```